### PR TITLE
Error handling enhancements

### PR DESCRIPTION
**Pull request by yosriz**
* add new exceptions classes : TransactionFailedException,NoKinTrustException, AccountNotFoundException
* change EthereumClientException to ClientException
* handle no account and no kin trust-line in transaction and balance
* wrap all specific Stellar (non-network) errors with TransactionFailedException
* handle wrong public address format errors (wrap with OperationFailedException)

### CODE REVIEW

File: kin-sdk-core/src/main/java/kin/sdk/core/ClientWrapper.java [hunk 1]
```diff
@@ -44,11 +47,11 @@ private Server initServer() {
         return new Server(serviceProvider.getProviderUrl());
     }
 
-    private KeyStore initKeyStore() throws EthereumClientException {
+    private KeyStore initKeyStore() throws ClientException {
```
Comment (yosriz) [inline kin-sdk-core/src/main/java/kin/sdk/core/ClientWrapper.java]: remove ClientException, add later if needed

File: kin-sdk-core/src/main/java/kin/sdk/core/ClientWrapper.java [hunk 2]
```diff
@@ -80,11 +85,17 @@ Balance getBalance(Account account) throws OperationFailedException {
                     break;
                 }
             }
+        } catch (HttpResponseException httpError) {
+            if (httpError.getStatusCode() == 404) {
```
Comment (yosriz) [inline kin-sdk-core/src/main/java/kin/sdk/core/ClientWrapper.java]: add comment to describe 404 stellar error

File: kin-sdk-core/src/main/java/kin/sdk/core/KinAccount.java [hunk 3]
```diff
@@ -23,7 +26,7 @@
 
     /**
      * Create {@link Request} for signing and sending a transaction of the given amount in kin to the specified public
-     * address Ethereum gas will be handled internally.
+     * address
```
Comment (yosriz) [inline kin-sdk-core/src/main/java/kin/sdk/core/KinAccount.java:29]: add reference to exceptions list in sync version

File: kin-sdk-core/src/main/java/kin/sdk/core/TransactionSender.java [hunk 4]
```diff
@@ -27,25 +33,29 @@
...
     @NonNull
     TransactionId sendTransaction(@NonNull Account from, String passphrase, @NonNull String publicAddress,
         @NonNull BigDecimal amount)
         throws OperationFailedException, PassphraseException {
 
         checkAddressNotEmpty(publicAddress);
-        checkForNegativeAMount(amount);
-        KeyPair addressee = KeyPair.fromAccountId(publicAddress);
+        checkForNegativeAmount(amount);
+        KeyPair addressee = validateAndGetAddresseeKeyPair(publicAddress);
```
Comment (yosriz) [inline kin-sdk-core/src/main/java/kin/sdk/core/TransactionSender.java]: generateAddresseeKeyPair

File: kin-sdk-core/src/main/java/kin/sdk/core/TransactionSender.java [hunk 5]
```diff
@@ -27,25 +33,29 @@
...
         verifyPayToAddress(addressee);
         KeyPair secretSeedKeyPair = keyStore.decryptAccount(from, passphrase);
         AccountResponse sourceAccount = loadSourceAccount(secretSeedKeyPair);
+        checkKinTrust(sourceAccount);
```
Comment (yosriz) [inline kin-sdk-core/src/main/java/kin/sdk/core/TransactionSender.java]: move `checkKinTrust` inside `loadSourceAccount`

File: kin-sdk-core/src/main/java/kin/sdk/core/TransactionSender.java [hunk 6]
```diff
@@ -77,37 +97,43 @@ private void verifyPayToAddress(KeyPair addressee) throws OperationFailedExcepti
...
         }
 
-        // Second, check the addressee has trustline with kinIssuer.
-        if (checkKinTrust(addresseeAccount)) {
-            throw new OperationFailedException("Addressee don't have Kin asset trust");
+        checkKinTrust(addresseeAccount);
+    }
+
+    private void checkKinTrust(AccountResponse accountResponse) throws NoKinTrustException {
+        if (!doesHasKinBalance(accountResponse)) {
```
Comment (yosriz) [inline kin-sdk-core/src/main/java/kin/sdk/core/TransactionSender.java]: `doesHasKinBalance` -> `hasKinBalance`

File: kin-sdk-core/src/main/java/kin/sdk/core/TransactionSender.java [hunk 7]
```diff
@@ -77,37 +97,43 @@ private void verifyPayToAddress(KeyPair addressee) throws OperationFailedExcepti
...
+            throw new NoKinTrustException(accountResponse.getKeypair().getAccountId());
         }
     }
 
     private AccountResponse loadSourceAccount(@NonNull KeyPair from) throws OperationFailedException {
         AccountResponse sourceAccount;
         try {
             sourceAccount = server.accounts().account(from);
+        } catch (HttpResponseException httpError) {
```
Comment (yosriz) [inline kin-sdk-core/src/main/java/kin/sdk/core/TransactionSender.java]: common code with `verifyPayToAddress`

File: kin-sdk-core/src/main/java/kin/sdk/core/KinAccount.java (hunk 3 above)
Comment (yosriz) [reply kin-sdk-core/src/main/java/kin/sdk/core/KinAccount.java:29]: done

**Review by berryve — APPROVED** [review]

**Review by berryve — APPROVED** [review]

**Comment by berryve** [comment]
I approved but would prefer that you move the enum back up + your tests fail

**Comment by yosriz** [comment]
@berryve fixed enum, tests will be handled separately

**Pull request merged**